### PR TITLE
Added QLoRA support in NeuralChat finetuning and refined NeuralChat optimization API.

### PR DESCRIPTION
**Pull request by XinyuYe-Intel**
## Type of Change

Added QLoRA support in NeuralChat finetuning and refined NeuralChat optimization API.
API changed

## Description

Added QLoRA support in NeuralChat finetuning and refined NeuralChat optimization API.
JIRA ticket: xxx

## Expected Behavior & Potential Risk

pass UT

## How has this PR been tested?

UT

## Dependency Change?

no library dependency introduced or removed

### CODE REVIEW

**Comment by hshen14** [comment]
@XinyuYe-Intel  please rebase and migrate your changes. Thanks.

**Comment by XinyuYe-Intel** [comment]
> @XinyuYe-Intel please rebase and migrate your changes. Thanks.

Done.

File: intel_extension_for_transformers/neural_chat/config.py [hunk 1]
```diff
@@ -209,7 +209,7 @@ class DataArguments:
         metadata={"help": "The list of special tokens to add in tokenizer."}
     )
     max_source_length: Optional[int] = field(
-        default=512,
+        default=384,
```
Comment (hshen14) [inline intel_extension_for_transformers/neural_chat/config.py:212]: why 384?

File: intel_extension_for_transformers/neural_chat/config.py [hunk 2]
```diff
@@ -218,14 +218,26 @@ class DataArguments:
         },
     )
     max_target_length: Optional[int] = field(
-        default=256,
+        default=128,
         metadata={
             "help": (
                 "The maximum total sequence length for target text after tokenization. Sequences longer "
```
Comment (hshen14) [inline intel_extension_for_transformers/neural_chat/config.py:224]: Why 128?

**Review by VincyZhang — APPROVED** [review]

**Review by lvliang-intel — APPROVED** [review]

**Pull request merged**